### ISSUE TITLE
[RFC] Prototyping speed improvements

### ISSUE THREAD

**Issue by AnneKitsune** [amethyst/rfcs#18]
## Problem
Prototyping in amethyst is slow.

100 lines of code seems to be the minimum to make the smallest of game. While its not a lot, it definitely is more than necessary.

## Source
I'm taking https://github.com/amethyst/amethyst/blob/develop/examples/sphere/main.rs as a model
Lack of good defaults for generic types.
Imports
Lack of handle_event utils for simple cases -> Quit on any key, quit on some key, is key pressed.

## Propositions
>Lack of good defaults for generic types.

Adding sensible defaults as type alias for generics.
1) Add a default.rs file.
2) Make a DefaultSomething type alias with a default consistent with the others.
3) Export the module default.

Example:
amethyst_input/src/default.rs
```
type DefaultInputHandler = InputHandler<String,String>
```

>Imports

Expand the prelude to include as many common types as possible. I'm not too sure how much this slows down compile time when using the prelude, but I do have performance degradation in one of my project where I was using only preludes.

>Lack of handle_event utils for simple cases -> Quit on any key, quit on some key, is key pressed.

I already implemented it inside of a custom project, currently debating if this should be inside of the engine.

**Comment by Xaeroxe** [comment]
I agree there shouldn't be a function for everything, but this qualifies as extremely common.  I say we should add it.

**Comment by torkleyy** [comment]
@AlmightyScientist Yes, this issue did not change a lot. I'll go through all the RFCs soon and make sure they're all discussed properly.
